src/backend/distributed/commands/foreign_server.c

```diff
@@ -230,6 +231,14 @@ PostprocessCreateForeignServerStmt(Node *node, const char *queryString)
 
 	const bool missingOk = false;
 	ObjectAddress address = GetObjectAddressFromParseTree(node, missingOk);
+
+	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&address);
+	if (errMsg != NULL)
+	{
+		RaiseDeferredError(errMsg, WARNING);
+		return NIL;
```

Comment:
Can you add a test to cover that part?

Reply:
and also for create extension, or maybe drop both changes ?

Reply:
dropped both